kmod/src/lock.c

```diff
@@ -667,7 +667,6 @@ static void lock_grant_worker(struct work_struct *work)
 		lock->request_pending = 0;
 		lock->mode = nl->new_mode;
 		lock->write_version = le64_to_cpu(nl->write_version);
-		lock->roots = gr->roots;
```

Comment:
Keep going, its a chain of single uses being removed:
 - go back to grant_reply using net_lock struct, remove grant_reply struct
 - lock_server doesn't need to fill grant roots, so scoutfs_server_get_fs_roots can go back to just being a private get_fs_roots (to match set_fs_roots) which is only called by the message handler.

You can see this stuff being introduced in 
```
commit b7943c5412ba7995f6205fac07cedf36a1fc652f
Author: Zach Brown <zab@versity.com>
Date:   Thu May 7 11:21:27 2020 -0700

    scoutfs: avoid reading forest roots with block IO

```